automation_tools/scripts/ga-migration/main.py

```diff
@@ -0,0 +1,238 @@
+
+
+def file_contains(term, filepath):
+    """Check whether file contains given term."""
+    with open(filepath) as f:
+        return term in f.read()
+
+
+def append_to_file(text, filepath):
+    """Append text to file."""
+    with open(filepath, "a") as f:
+        f.write(text)
+
+
+def add_line(term, filepath):
+    """ Add a line to a file """
+    logging.info("TASK: Adding line '%s' to %s" % (term, filepath))
+    # If the file exists
+    if os.path.isfile(filepath):
+        # And the line is not already there
+        if not file_contains(term, filepath):
+            append_to_file(term, filepath)
+        else:
+            logging.info("SKIPPED TASK. Line already there. ")
+    else:
+        logging.info("SKIPPED TASK. No %s found" % filepath)
+
+
+def replace_simple(text, replacing, filepath):
+    """
+    Replaces every match of a string with another in the specified file
+    """
+    logging.info(
+        "TASK: Simple replacing %s with %s in %s" % (text, replacing, filepath)
+    )
+    if os.path.isfile(filepath):
+        logging.info("Found %s" % filepath)
+        # TODO: expose number of matches
+        with fileinput.FileInput(filepath, inplace=True, backup=".bak") as file:
+            for line in file:
+                print(line.replace(text, replacing), end="")
+    else:
+        logging.info("No %s found" % filepath)
+
+
+def replace_regex(regex, output, filepath):
+    """
+    Replaces every match of a string with another in the specified file
+    """
+    logging.info(
+        "TASK: RegEx replacing %s with %s in %s" % (regex, output, filepath)
+    )
+    if os.path.isfile(filepath):
+        logging.info("Found %s" % filepath)
+        # TODO: expose number of matches
+        with fileinput.FileInput(filepath, inplace=True, backup=".bak") as file:
+            for line in file:
+                print(re.sub(regex, output, line), end="")
+    else:
+        logging.info("SKIPPED TASK. No %s found" % filepath)
+
+
+def download_file(url, destination):
+    # Get path
+    dirname = os.path.dirname(os.path.realpath(destination))
+    if not os.path.exists(dirname):
+        os.makedirs(dirname)
+    print(dirname)
+    r = requests.get(url, allow_redirects=True)
+    open(destination, "wb").write(r.content)
+
+
+def replace_list(filepath, regex, to_remove, to_add, var_name):
+    with open(filepath, "r") as f:
+        contents = f.read()
+
+    m = re.search(regex, contents)
+
+    matched_list_str = m.group(1)
+
+    parsedlist = ast.literal_eval(matched_list_str)
+
+    for element_to_remove in to_remove:
+        for element in parsedlist:
+            if element_to_remove in element:
+                parsedlist.remove(element)
+
+    for element in to_add:
+        parsedlist.append(element)
+
+    py_parsed_string = f"{var_name} = {json.dumps(parsedlist)}"
+
+    content2 = contents.replace(m.group(0), py_parsed_string)
+
+    with open(filepath, "w") as f:
+        f.write(content2)
```

Comment:
I noticed that the final result is something like:

```python
tests_require = ["check-manifest>=0.25", "coverage>=4.0", "isort>=4.2.2", "mock>=1.3.0", "pydocstyle>=1.0.0", "pytest>=4.0.0,<5.0.0", "pytest-invenio>=1.4.0"]
```
Not indented, this might cause problems with `pycodestyle` checks. Is it possible to add a break line after each dependency?

Reply:
```console
$ pycodestyle setup.py 
setup.py:34:80: E501 line too long (215 > 79 characters)
```

Reply:
Should be addressed by https://github.com/inveniosoftware/automation-tools/pull/2/commits/c9ab66981ed40ce40012f47e9b8f51df10fa1acb